### PR TITLE
xflux-gui: init at 1.1.1

### PR DESCRIPTION
**Pull request by sheenobu**
###### Things done
- [ ] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
  - [x] NixOS
  - [ ] OS X
  - [ ] Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by sheenobu** [comment]
xflux itself is proprietary but xflux-gui is not. However, xflux-gui does ship its own version of the xflux binary, which I discard. Not sure if that means we have to mark this as unfree or not.

File: pkgs/tools/misc/xflux/gui.nix [hunk 1]
```diff
@@ -0,0 +1,42 @@
+{ stdenv, pkgs, fetchFromGitHub, pythonPackages }:
```
Comment (joachifm) [inline pkgs/tools/misc/xflux/gui.nix]: Please pass in specific packages rather than the entire package set.

**Comment by joachifm** [comment]
The GUI output will end up with a reference to the unfree `xflux`, thus it would be illegal to redistribute the resulting closure (assuming it's illegal to redistribute `xflux` itself). Is it feasible to allow the GUI to resolve the `xflux` command from `PATH` instead?

**Comment by sheenobu** [comment]
@joachifm Good point, thanks.

The github repo for xflux-gui contains a copy of the xflux binary (itself non-free) and while I'm not using it, does that mean the entire github repo is non-free and shouldn't be cached by the nixos cache? I think I might have to mark the whole thing as 'nonfree'. Thoughts?

File: pkgs/tools/misc/xflux/gui.nix (hunk 1 above)
Comment (sheenobu) [inline pkgs/tools/misc/xflux/gui.nix]: Took care of this. Thank you.

**Comment by joachifm** [comment]
That's unfortunate, and a little odd. Perhaps the binaries *can* be redistributed? The only thing I could find is the EULA, which clearly states that redistribution of f.lux is prohibited, but I'm not sure how that meshes with embedding binaries in a so-called open source repo on github ... Maybe ask the `xflux-gui` devs what they think?

I'd mark all of it unfree for now, just to be safe.

**Comment by sheenobu** [comment]
I marked it all as unfree for now, so it should be ready to merge. I'll ask the xflux-gui devs and may go into IRC to ask about the philosophical issues of a free program calling out to a non-free one.

**Comment by joachifm** [comment]
Thanks

**Pull request merged**